### PR TITLE
Include Telemetry Version

### PR DESCRIPTION
**Pull request by jbinz**

### CODE REVIEW

File: firmware/src/tasks/task_telemetry.cpp [hunk 1]
```diff
@@ -296,8 +303,21 @@ void Telemetry::SendSettings(uint8_t command, uint8_t value) const noexcept {
   HAL_UART_Transmit(&TELEMETRY_UART_HANDLE, out, 4, 2);
 }
 
+void Telemetry::RequestVersionNum() const noexcept {
+  /* Return if the code version was fetched */
+  if (static_cast<bool>(strlen(telemetry_code_version))) {
```
Comment (stojadin2701) [inline firmware/src/tasks/task_telemetry.cpp:308]: if (strlen(telemetry_code_version) > 0)

File: firmware/src/tasks/task_telemetry.cpp [hunk 2]
```diff
@@ -267,7 +268,13 @@ bool Telemetry::Parse(uint8_t op_code, const uint8_t* buffer, uint32_t length, g
     if (m_amplifier_temperature > k_amplifier_hot_limit) {
       add_error(CATS_ERR_TELEMETRY_HOT);
     }
-    //    log_raw("Got temp %f", static_cast<double>(m_amplifier_temperature));
+  } else if (op_code == CMD_VERSION_INFO) {
+    uint8_t version_buffer[length];
+    memcpy(&version_buffer[0], buffer, length);
+    for (uint8_t i = 0; i < length; i++) {
+      telemetry_code_version[i] = static_cast<char>(version_buffer[i]);
+    }
```
Comment (stojadin2701) [inline firmware/src/tasks/task_telemetry.cpp:276]: can we do strncpy here?

**Comment by stojadin2701** [comment]
Address this from clang-tidy comments:

image

File: firmware/src/config/globals.cpp [hunk 3]
```diff
@@ -56,3 +56,5 @@ volatile recorder_status_e global_recorder_status = REC_OFF;
 event_action_map_elem_t* event_action_map = nullptr;
 
 const char* code_version = "2.3.2";
+char telemetry_code_version[20];
```
Comment (stojadin2701) [inline firmware/src/config/globals.cpp:59]: initialize it to 0

```suggestion
char telemetry_code_version[20] = {};
```

**Review by l-jost — APPROVED** [review]

**Pull request merged**